docs/contributor/pocs/gomemlimit-telemetry-memory.md

```diff
@@ -0,0 +1,69 @@
+# GOMEMLIMIT for Telemetry Components
```

Comment:
This whole document feels more like marketing than technical writing or one of our usual PoCs. Where is this content from? I don't recognize the writing style, is it from Go product marketing? There's a lot of fluff, it could be shortened considerably. 

What's the purpose of this document? Are we evaluating whether we want to use GOMEMLIMIT? If yes, what's the problem we want to solve with it? Do I need to read the whole big document to understand the basic premise?